### PR TITLE
loss_barlow_twins: add get_eccm member function

### PR DESCRIPTION
**Pull request by arrufat**
This allows us to greatly simplify the self supervised learning example:
- the computation in user code was a bit too distracting
- avoids duplicated computation/allocation of this matrix
- avoids edge case where net outputs are zero due to trainer synchronization

### CODE REVIEW

File: dlib/dnn/loss_abstract.h [hunk 1]
```diff
@@ -2144,6 +2144,20 @@ namespace dlib
                   in WHAT THIS OBJECT REPRESENTS for details.
         !*/
 
+        tensor& get_eccm();
+        /*!
+            ensures
+                - returns the empirical cross-correlation matrix computed by the loss.
+                - this is only meant to be used for visualization/debugging purposes.
+        !*/
```
Comment (arrufat) [inline dlib/dnn/loss_abstract.h:2152]: I am not sure if this will ever be used. But even if the user modifies the `eccm`, it doesn't really affect the loss, since it's re-computed at every step.

So, in case someone has to modify the `eccm` between training steps, they don't need to copy it.

**Comment by davisking** [comment]
Yeah, nice :D
